src/server/player_sao.cpp

```diff
@@ -519,12 +519,14 @@ void PlayerSAO::rightClick(ServerActiveObject *clicker)
 
 void PlayerSAO::setHP(s32 target_hp, const PlayerHPChangeReason &reason, bool from_client)
 {
+	s32 hp_change = std::max<s32>(target_hp, -U16_MAX) - (s32)m_hp;
```

Comment:
If the only goal here is overflow protection, wouldn't `S32_MIN + U16_MAX` be a more correct minimum value?